meta-mbl-distro/scripts/create-update-payload

```diff
@@ -97,128 +103,279 @@ def parse_args():
+
+    for elem in image_boot_files.split():
+        output_map[MBL_BOOT_ID] += img_boot_files_val_to_map_val(
+            elem, deploy_dir
         )
 
-    for var, fname in filename_var_map.items():
-        src = deploy_dir_img / fname
-        dest = staging_dir / fname
-        shutil.copy2(str(src), str(dest))
+    return output_map
+
+
+def copy_files_to_subdir(paths, dst_dir):
+    """
+    Copy files to a destination directory, renaming them when specified.
+
+    :param paths list<Path|tuple<Path, Path>>: a list of specifications of
+    files to copy to a destination directory. Each element can just be a
+    plain Path, in which case it is copied to the destination directory with
+    its existing filename; otherwise the element should be a 2-tuple of Paths
+    (src_path, dst_name) where src_path will be copied to dst_dir with the name
+    dst_name.
+
+    :param dst_dir Path: the directory into which files will be copied.
+    """
+    for path in paths:
+        if isinstance(path, tuple):
+            src_path = path[0]
+            dst_name = path[1].name
+        else:
+            src_path = path
+            dst_name = path.name
+        # copy2 preserves file metadata and follows symlinks
+        shutil.copy2(str(src_path), str(dst_dir / dst_name))
+
+
+def copy_single_file_modifying_name(path, new_name, dst_dir):
+    if isinstance(path, list) and len(path) > 1 or not isinstance(path, list):
+        raise ValueError("Expects a list with a single element.")
+
+    for file_path in path:
+        shutil.copy2(str(file_path), str(dst_dir / new_name))
 
 
-def create_payload_tar_archive(tar_archive_name, dir_path, fname_var_map):
+def copy_to_staging_dir(filename_var_map, staging_dir, tinfoil=None):
+    """
+    Copy files to a staging dir and prepend the file name with the bb varname.
+
+    :param filename_var_map dict: Mapping of bitbake variable names to files.
+    :param staging dir Path: path to the dir where the files will be copied to.
+    """
+    for var, paths in filename_var_map.items():
+        if var == MBL_BOOT_ID:
+            dst_dir = staging_dir / var
+            dst_dir.mkdir(exist_ok=True)
+            copy_files_to_subdir(paths, dst_dir)
+        else:
+            dst_dir = staging_dir
+            copy_single_file_modifying_name(paths, var, dst_dir)
+
+def add_payload_format_version_to_tar_archive(tar):
+    buf = io.BytesIO(PAYLOAD_FORMAT_VERSION)
+    tar_info = tarfile.TarInfo(name="payload_format_version")
+    tar_info.size = len(buf.getvalue())
+    tar_info.mode = 0o444
+    tar_info.uid = 0
+    tar_info.gid = 0
+    tar_info.uname = "root"
+    tar_info.gname = "root"
+    tar_info.mtime = time.time()
+    tar.addfile(tar_info, buf)
+
+def create_tar_archive(tar_archive_name, dir_path, fname_var_map):
     """
-    Create the payload tar archive.
+    Create a tar archive.
 
     The tar archive is uncompressed but the files within it are compressed
-    using xz.
+    using gz.
```

Comment:
Why switch from xz to gz? IIUC, xz tends to give much better compression at the cost of more computation on the compressing side, but not much more computation on the decompressing side, so it seems perfect for IoT.

Reply:
Changed to use `xz` in ac01602